tests/test_types.py

```diff
@@ -104,3 +111,49 @@ def testNonAscii(self):
         # this should work with either a unicode or a bytes
         self.assertEqual(map_username_to_mxid_localpart("têst"), "t=c3=aast")
         self.assertEqual(map_username_to_mxid_localpart("têst".encode()), "t=c3=aast")
+
+
+class UserTestCase(unittest.TestCase):
```

Comment:
I think this is mostly testing that `attrs` works as expected? I don't think I'd bother.